s3transfer/upload.py

```diff
@@ -286,6 +286,157 @@ def _get_put_object_fileobj(self, fileobj):
         return fileobj
 
 
+class UploadNonSeekableInputManager(UploadInputManager):
+    """Upload utility for a file-like object that cannot seek."""
+    def __init__(self, osutil, transfer_coordinator):
+        super(UploadNonSeekableInputManager, self).__init__(
+            osutil, transfer_coordinator)
+        self._initial_data = None
+
+    @classmethod
+    def is_compatible(cls, upload_source):
+        return not seekable(upload_source)
```

Comment:
So I am not sure if the qualifier here should be weather it is seekable or not. Technically, you could put a seekable file through this Manager and it should work. I think the compatibility here should be if the file is readable as if it fails that the ``UploadSubmissionTask`` will throw an error as this is the last one in the input manager resolver chain.